crates/anvil/src/eth/api.rs

```diff
@@ -78,6 +78,25 @@ use futures::channel::{mpsc::Receiver, oneshot};
 use parking_lot::RwLock;
 use std::{collections::HashSet, future::Future, sync::Arc, time::Duration};
 
+macro_rules! prepare_typed_tx {
+    ($self: expr, $request:expr, $typed_tx:ident, $from:ident, $nonce:ident, $on_chain_nonce:ident) => {
+        let $from = $request.from.map(Ok).unwrap_or_else(|| {
+            $self.accounts()?.first().cloned().ok_or(BlockchainError::NoSignerAvailable)
+        })?;
+
+        let ($nonce, $on_chain_nonce) = $self.request_nonce(&$request, $from).await?;
+
+        if $request.gas.is_none() {
+            // estimate if not provided
+            if let Ok(gas) = $self.estimate_gas($request.clone(), None, None).await {
+                $request.gas = Some(gas.to());
+            }
+        }
+
+        let $typed_tx = $self.build_typed_tx_request($request, $nonce)?;
+    };
+}
```

Comment:
please remove this macro, this is very hard to read and unclear why this is needed.

you should be able to use

https://github.com/alloy-rs/alloy/blob/143f175d934b83619b02ae6dd820f5e145c64381/crates/rpc-types-eth/src/transaction/mod.rs#L259-L259

instead

Reply:
well, actually it removes a lot of unnecessary code duplication in `EthApi`. Maybe I should think about its realisation more and make another PR...

But for now, I have removed it